java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -522,6 +615,23 @@ protected <T> T handleValkeyResponse(
             return null;
         }
 
+        // Handle DirectByteBuffer conversion for large responses (>16KB)
+        if (value instanceof java.nio.ByteBuffer) {
+            java.nio.ByteBuffer buffer = (java.nio.ByteBuffer) value;
+            byte[] bytes = new byte[buffer.remaining()];
+            buffer.get(bytes);
+            if (classType == String.class && encodingUtf8) {
+                value = new String(bytes, java.nio.charset.StandardCharsets.UTF_8);
```

Comment:
It looks like we can use StandardCharsets.UTF_8.decode(<ByteBuffer>) instead of copying to a byte array.